src/Disks/S3/registerDiskS3.cpp

```diff
@@ -145,9 +145,18 @@ void registerDiskS3(DiskFactory & factory)
             config.getUInt64(config_prefix + ".min_bytes_for_seek", 1024 * 1024));
 
         /// This code is used only to check access to the corresponding disk.
-        checkWriteAccess(*s3disk);
-        checkReadAccess(name, *s3disk);
-        checkRemoveAccess(*s3disk);
+        try
+        {
+            checkWriteAccess(*s3disk);
+            checkReadAccess(name, *s3disk);
+            checkRemoveAccess(*s3disk);
+        }
+        catch (...)
```

Comment:
It's better to specify explicit exception type. What if the problem is in another place instead of S3 unavailability?

Reply:
At the moment it's hard to distinguish S3-related exceptions.
I'll rewrite this piece of code in another way.